sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3167,6 +3167,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val LEGACY_CAST_TO_STRING =
+    buildConf("spark.sql.legacy.castToString.enabled")
```

Comment:
Do we plan to drop the legacy string format in the future?

Reply:
If we want to make it consistent, we should add the configuration to `Cast` too I think.